server.go

```diff
@@ -168,15 +180,36 @@ func (server *Server) handle(s *streamWrap) error {
 
 	replyv = reflect.New(mtype.ReplyType.Elem())
 
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	ctxv := reflect.ValueOf(ctx)
+
+	// This is a connection watchdog. We do not
+	// need to read from this stream anymore.
+	// However we'd like to know if the other side is closed
+	// (or reset). In that case, we need to cancel our
+	// context. Note this will also happen at the end
+	// of a successful operation when we close the stream
+	// on our side.
+	go func() {
```

Comment:
@stebalian: I don't know if this is too hacky. Is there a better way to detect a premature death of a stream? (stream being closed or reset-mostly from the other side)

Reply:
In short, no (other than writing and failing). This is actually the correct way to handle this.

However, I assume this is a one-way channel, right? In that case, shouldn't you reset the stream here if you either read data (the other side shouldn't send any) or see an error that's not an EOF?

Reply:
Yeah, that won't hurt and hardens things a bit..